sms_sponsorship/forms/sms_sponsorship_registration_form.py

```diff
@@ -209,7 +209,24 @@ def form_after_create_or_update(self, values, extra_values):
                 _logger.error("Activate sponsorship is not yet implemented")
 
             # update sms request
-            sms_request.complete_step2()
+            self.env['sms.child.request'].sudo()\
```

Comment:
This is not good, you reverted some of the code that I refactored. Please rebase your branch onto devel and verify that you don't revert the code.